### ISSUE TITLE
`normalizationConfig` should control `dbt` transformation options (for Destinations V2)

### ISSUE THREAD

**Issue by evantahler** [airbytehq/airbyte#28325]
# What needs to be done?

This story is to determine how the Airbyte UI and API should change once in-container normalization is removed.  I *think* we only need to remove the `Normalization` section of the page, but leave the `Transformation` and `Custom Transformation` sections as they are. 

I believe that:
* `normalizationConfig` is about in-container normalization
* `supportsDbt` is about external dbt usage, either dbt cloud (on Airbyte Cloud) or a custom dbt container (on Airbyte OSS)
* The two options are actually independent of each other

The definition of done for this story is to confirm this behavior:
* Pick a destination that does not have `normalizationConfig` or `supportsDbt` (e.g. Teradata) and enable _just _ `supportsDbt`
* Build a local OSS registry and run the OSS app
* See what happens and what the UI does.

# Current Behavior

### For Cloud:
Today, the `supportsDbt` metadata option configures the existence of both in-container and dbt-cloud normalization.  Since in-container Normalization is going away, this same option can continue to be used _only_ for dbt cloud support.  

**BigQuery** (supportsDbt=true)

<img width="1627" alt="Screenshot 2023-07-14 at 10 20 44 AM" src="https://github.com/airbytehq/airbyte/assets/303226/f34c8ac5-a40f-4cca-8291-19a2403f56b3">

**Google Sheets** (supportsDbt=false)

<img width="1627" alt="Screenshot 2023-07-14 at 10 20 23 AM" src="https://github.com/airbytehq/airbyte/assets/303226/36f4c6d0-90ec-4485-b723-a766a7034b0f">

### For OSS:

**BigQuery** (supportsDbt=true)

<img width="1509" alt="Screenshot 2023-07-14 at 10 40 28 AM" src="https://github.com/airbytehq/airbyte/assets/303226/1e8aa1ea-02dd-4ac8-ae91-a2958c8d6547">

<img width="1509" alt="Screenshot 2023-07-14 at 10 39 45 AM" src="https://github.com/airbytehq/airbyte/assets/303226/c3c10b75-27b3-4cdd-8e3a-a2bf1d842466">

**Google Sheets** (supportsDbt=false)

Same as Cloud

**Comment by evantahler** [comment]
Related slack converstation https://airbytehq-team.slack.com/archives/C03VDJ4FMJB/p1689357262800909

**Comment by evantahler** [comment]
https://github.com/airbytehq/airbyte/issues/28388 implies that there may be a bug with this logic, or something incorrect with the above assumption

**Comment by evantahler** [comment]
@timroes is it maybe the case that these controls are entirely in the front-end?

**Comment by timroes** [comment]
@evantahler could you clarify what exactly you mean with "the controls". Whether we show this form or not is determined in the FE. We can adjust that as needed for the new normalization, so I feel we should just define how we want it to work and then can make the appropriate adjustments.

**Comment by evantahler** [comment]
Hmm, @davinchia thought that maybe the front-end had some control over the display.  If that's not the case, then it is probably likely that the server isn't quite doing the right thing.  

The desired state is:
* the "Normalization" option appears only if the connector's metadata includes `normalizationConfig`
* The "Custom Transformation" (OSS) or "Transformation" (Cloud) option appears only if the connector's metadata includes `supportsDbt=true`

**Comment by tealjulia** [comment]
I think the Platform is doing something funky somewhere between getting the `metadata.yaml` and returning the destination definition to the frontend.  

I would expect the case outlined above to completely remove the `normalizationConfig` object from the response.  I intercepted the API call locally and removed that property, and with a little finessing (~2 line fix due to our auto-generated types saying that property is required), the UI rendered as I would expect:

- the dbt transformations sections always showed up if `supportsDbt=true` (and the flag for whether or not it's allowed was true)
- the normalization section did _not_ show up

This makes me think that the API is not removing the property (nor setting `supported` to `false`, which also yielded the expected results)

[Examples and screenshots in this Slack thread](https://airbytehq-team.slack.com/archives/C03088BTMFC/p1690392420659979)

**Comment by evantahler** [comment]
cc @davinchia  Note: Temporal workflow also needs to know not to run dbt/normalization when `normalizationConfig` is false/null.  That's probably more important that the UI doing the right thing ;)

**Comment by evantahler** [comment]
I did some more research into this, and at least now, the UI does seem to be properly responding the to `supportsDbt` and (normalization_repository + normalization_tag + normalization_integration_type) columns in the `actor_definition_version` table.  I cannot reproduce this bug any more. If the values are null, the option goes away in the UI as expected. 

<img width="938" alt="Screenshot 2023-08-11 at 4 20 40 PM" src="https://github.com/airbytehq/airbyte/assets/303226/9d1e16b7-cb1d-4478-9700-f2c3e278d6a7">  Change these values in the DB and the UI changes as well.  If there is a problem, it is isolated to setting proper values on the ADV from metadata/connector-catalogs.


I also ran a sync that used to have normalization enabled, NULL'd out the normalization columns on the actor_definition_version, and synced again.  Here there is a bug:

```
Caused by: java.lang.NullPointerException: Cannot invoke "String.split(String, int)" because the return value of "io.airbyte.persistence.job.models.IntegrationLauncherConfig.getNormalizationDockerImage()" is null
        at io.airbyte.workers.temporal.sync.NormalizationActivityImpl.getNormalizationImageTag(NormalizationActivityImpl.java:240) ~[io.airbyte-airbyte-workers-dev.jar:?]
        at io.airbyte.workers.temporal.sync.NormalizationActivityImpl.normalizationSupportsV1DataTypes(NormalizationActivityImpl.java:230) ~[io.airbyte-airbyte-workers-dev.jar:?
```

I've fixed that up via https://github.com/airbytehq/airbyte-platform-internal/pull/8314

I'm going to close this issue as we cannot reproduce it.
